src/lib389/cli/dscontainer

```diff
@@ -425,6 +425,12 @@ def begin_healthcheck(ds_proc, log_exception):
     return (True, False)
 
 
+def stop():
+    pid = pid_from_file(PID_FILE)
+    os.kill(pid, signal.SIGTERM)
+    os.waitpid(pid)
```

Comment:
When I run `/usr/libexec/dirsrv/dscontainer --stop`, it stops the DS, but throws an error and exits immediately instead of waiting:
```
podman exec -til /bin/bash
bash-5.2# /usr/libexec/dirsrv/dscontainer --stop
Traceback (most recent call last):
  File "/usr/libexec/dirsrv/dscontainer", line 485, in <module>
    stop()
  File "/usr/libexec/dirsrv/dscontainer", line 431, in stop
    os.waitpid(pid)
TypeError: waitpid() takes exactly 2 arguments (1 given)
```
`os.waitpid` can't be used here because it's only applicable for the child processes. It's used in another place in `dscontainer` but only because `dscontainer` actually starts that process that we wait for.

Instead we can check if PID exists by using lib389 function [`pid_exists`](https://github.com/389ds/389-ds-base/blob/1f95b57fb8b03f72c7d2a2ce5b14eed5ba320407/src/lib389/lib389/__init__.py#L181):

```suggestion
    count = 10 # Instead of hardcoding maybe we can add a similar env variable like DS_STARTUP_TIMEOUT
    while pid_exists(pid) and count > 0:
        time.sleep(1)
        count -= 1
    if pid_exists(pid):
        os.kill(pid, signal.SIGKILL)
```

Reply:
Sounds good.